### DISCUSSION TITLE
Multi threaded safe?

### DISCUSSION THREAD

**Discussion by conormdowney** [diesel-rs/diesel#3116] [Q&A]
I cant seem to see it mentioned anywhere but is diesel safe to use in a multi threaded app? If I create a single connection can that be cloned to threads and used by multiple?

**Comment by weiznich** [comment]
Yes using diesel is safe in multithreaded apps. That's not explicitly documented as that's something that rust guarantees by default. You likely want to use a database pool in such an environment. Checkout [r2d2](https://docs.rs/r2d2/0.8.9/r2d2/) and [diesels r2d2 integration](https://docs.diesel.rs/1.4.x/diesel/r2d2/index.html) for this.
